File: validate/cc_corpus/fixtures/equivalence.go
package fixtures

// NoBranches exercises straight-line code (CC=1).
func NoBranches(x int) int {
	return x + 1
}

// SingleIf exercises one if/else (CC=2).
func SingleIf(x int) int {
	if x > 0 {
		return 1
	}
	return 0
}

// IfElseIf exercises if/else-if/else chain (CC=3).
func IfElseIf(x int) int {
	if x > 0 {
		return 1
	} else if x < 0 {
		return -1
	} else {
		return 0
	}
}

// NestedIf exercises if inside if (CC=3).
func NestedIf(x, y int) int {
	if x > 0 {
		if y > 0 {
			return 1
		}
	}
	return 0
}

// ForLoop exercises a simple for loop (CC=2).
func ForLoop(lst []int) int {
	for i := 0; i < len(lst); i++ {
	}
	return 0
}

// ForWithIf exercises a for loop containing an if (CC=3).
func ForWithIf(lst []int) int {
	for i := 0; i < len(lst); i++ {
		if lst[i] > 0 {
			return lst[i]
		}
	}
	return 0
}

// WhileLoop exercises a for loop used as while (CC=2).
func WhileLoop(n int) int {
	for n > 0 {
		n--
	}
	return n
}

// TryCatch is a stub — Go has no try/catch; skipped via expected.json.
func TryCatch() int {
	return 0
}

// BooleanOps exercises boolean operators in a condition (CC=4).
func BooleanOps(a, b, c bool) int {
	if a && b || c {
		return 1
	}
	return 0
}

// EarlyReturn exercises multiple guard clauses (CC=3).
func EarlyReturn(val int) int {
	if val < 0 {
		return -1
	}
	if val == 0 {
		return 0
	}
	return 1
}

// Ternary uses if/else since Go has no ternary operator (CC=2).
func Ternary(x bool, a, b int) int {
	if x {
		return a
	}
	return b
}

// SwitchCase exercises a switch with 2 cases + default (CC=4 in Go).
func SwitchCase(x int) int {
	switch x {
	case 1:
		return 10
	case 2:
		return 20
	default:
		return 0
	}
}

File: validate/cc_corpus/fixtures_cc/equivalence.c
int no_branches() {
    return 42;
}

int single_if(int x) {
    if (x > 0) {
        return 1;
    } else {
        return 0;
    }
}

int if_else_if(int x) {
    if (x > 0) {
        return 1;
    } else if (x < 0) {
        return -1;
    } else {
        return 0;
    }
}

int nested_if(int a, int b) {
    if (a > 0) {
        if (b > 0) {
            return 1;
        }
    }
    return 0;
}

int for_loop(int n) {
    int s = 0;
    for (int i = 0; i < n; i++) {
        s += i;
    }
    return s;
}

int for_with_if(int* items, int len) {
    int count = 0;
    for (int i = 0; i < len; i++) {
        if (items[i] > 0) {
            count++;
        }
    }
    return count;
}

int while_loop(int x) {
    int n = 0;
    while (x > 0) {
        x--;
        n++;
    }
    return n;
}

int try_catch() {
    // C has no try/catch — stub, skipped by skip_c in expected.json
    return 0;
}

int boolean_ops(int a, int b, int c) {
    if (a && b || c) {
        return 1;
    }
    return 0;
}

int early_return(int x) {
    if (x > 0) {
        return 1;
    }
    if (x < 0) {
        return -1;
    }
    return 0;
}

int ternary(int x) {
    return x > 0 ? 1 : 0;
}

int switch_case(int x) {
    switch (x) {
        case 1: return 1;
        case 2: return 2;
        default: return 0;
    }
}
